File: semestr-8/Programowanie obiektowe w Java/alexander/Java/Archiwum/2018/A/Builder/src/IdBuilder.java

import java.util.HashSet;

public class IdBuilder 
{
   private final HashSet<Long> lst;
   public IdBuilder()
   {
      lst=new HashSet<>();
   }
   public void clear()
   {
      lst.clear();
   }
   public void add(long idvalue)
   {
      lst.add(idvalue);
   }
   @Override public String toString() 
   {
      if(lst.size()<=0) return "null";
      final StringBuilder sb=new StringBuilder();
      for(Long idvalue:lst) sb.append(",").append(idvalue);
      return sb.toString().substring(1);
   }
}

File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2018/B/Builder/Builder/IdBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Builder
{
    class IdBuilder
    {
        private HashSet<long> list = new HashSet<long>();
        public string Delimeter { get; set; }
        public string EmptyValue { get; set; }
        public IdBuilder(string Delimeter=", ",string EmptyValue="null")
        {
            this.Delimeter = Delimeter;
            this.EmptyValue = EmptyValue;
        }
        public void Clear() { list.Clear(); }
        public void Add(long id) { list.Add(id); }
        public override string ToString()
        {
            if (list.Count <= 0) return "null";
            StringBuilder sb = new StringBuilder();
            /*
            bool first = true;
            foreach(long id in list)
            {
                if (first) first = false;
                else sb.Append(",");
                sb.Append(id);
            }
            return sb.ToString();
            */
            /*
            long first = list.First();
            foreach (long id in list)
            {
                if (id!=first) sb.Append(",");
                sb.Append(id);
            }
            return sb.ToString();
            */
            /*
            foreach (long id in list)
            {
                sb.Append(",").Append(id);
            }
            return sb.ToString().Substring(1);
            */
            return string.Join(",",list).ToString();
        }
    }
}
